pkg/restore/restore.go

```diff
@@ -1111,19 +1128,22 @@ func (ctx *restoreContext) restoreItem(obj *unstructured.Unstructured, groupReso
 		}
 	}
 
```

Comment:
I submitted a PR with these suggestions on the forked repository
https://github.com/RafaeLeal/velero/pull/1/files

I feel that's still complex, but it's hard to simplify without major refactorings.
In this suggested approach, you need to add and remove fields into an unstructured object, which I was trying to avoid... But I guess it's less intrusive. Let me know if that's what you intended

Reply:
"I guess it's less intrusive".  --- yes, agree. This is important I think, because we can avoid regressions.
"need to add and remove fields into an unstructured object" --- if you don't want to do this, I guess you can still copy the entire object but use the status field only. Anyway, let me review this with the team and get you back.
"I feel that's still complex". --- somehow better IMO, let me review with the team for any further improvement

Thanks a lot for your efforts.

Reply:
I have little context about this.. so I'm comfortable with both solutions, so I'd go with whatever the team feels it's better to maintain.
> This is important I think, because we can avoid regressions.
I think that if we are afraid of doing more intrusive changes, we should refactor the code so we can have better test coverage. I know that's an "easier said than done" kind of issue... but maybe we can come up with something

Reply:
@RafaeLeal 
Please help to merge https://github.com/RafaeLeal/velero/pull/1/files, since this make less change and less intrusive.
For the overall refactor to the existing logics, we may visit it during the data mover design & change, let's keep the existing logic as is for this PR.

@reasonerjt 
Please also help to have a look at the change where unstructured.NestedFieldCopy，unstructured.SetNestedField are used, see if you have any other proposal


Thanks.

Reply:
Merged 👍

Reply:
I think it look good.